provider/openstack/provider.go

```diff
@@ -648,6 +649,11 @@ func (e *Environ) Bootstrap(ctx environs.BootstrapContext, args environs.Bootstr
 func (e *Environ) supportsNeutron() bool {
 	client := e.client()
 	endpointMap := client.EndpointsForRegion(e.cloud.Region)
+	if len(endpointMap) == 0 {
+		// In some corner cases the endpoingMap is empty.  Default to
```

Comment:
I think you'll only get an empty map back if the client has not authenticated? Probably need to change the order of some code, so that auth happens first.